library/alloc/src/collections/binary_heap/mod.rs

```diff
@@ -434,8 +434,9 @@ impl<T: Ord> BinaryHeap<T> {
     /// heap.push(4);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_unstable(feature = "const_binary_heap_constructor", issue = "112353")]
     #[must_use]
-    pub fn new() -> BinaryHeap<T> {
+    pub const fn new() -> BinaryHeap<T> {
         BinaryHeap { data: vec![] }
```

Comment:
if you use `Vec::new()` here, then `BinaryHeap::new` can become const stable. `vec![]` goes through a code path that assumes it needs to allocate.

Reply:
Perhaps I should have marked onstification of `new` and `new_in` as different features, which may helps to stablize const `new` function.

Reply:
https://github.com/rust-lang/rust/blob/bbcc1691a4646a6d2ebb2189bdff5b878407fbe6/library/alloc/src/macros.rs#L42-L57

It seems that the macro `vec![]` (nothing filled within the bracket) just expands into `Vec::new()` **without allocator**.

Thus, constification of `BinaryHeap::new` is able to be stablized, if the `new` function does not share the mark `rust_const_unstable(feature = "const_binary_heap_constructor")` with `BinaryHeap::new_in`.